aws/resource_aws_kinesis_firehose_delivery_stream.go

```diff
@@ -653,32 +873,75 @@ func resourceAwsKinesisFirehoseDeliveryStreamCreate(d *schema.ResourceData, meta
 	return resourceAwsKinesisFirehoseDeliveryStreamRead(d, meta)
 }
 
+func validateAwsKinesisFirehoseSchema(d *schema.ResourceData) error {
+
+	_, s3Exists := d.GetOk("s3_configuration")
+	_, extendedS3Exists := d.GetOk("extended_s3_configuration")
+
+	if d.Get("destination").(string) == "extended_s3" {
+		if !extendedS3Exists {
+			return fmt.Errorf(
+				"When destination is 'extended_s3', extended_s3_configuration is required",
+			)
+		} else if s3Exists {
+			return fmt.Errorf(
+				"When destination is 'extended_s3', s3_configuration must not be set",
+			)
+		}
```

Comment:
if we include a `ConflictsWith` we can probably remove some of this validation above, I think